kernel/kernel-auth-service/src/main/java/io/mosip/kernel/auth/service/impl/ProxyAuthServiceImpl.java

```diff
@@ -0,0 +1,648 @@
+ * 
+ * @author Ramadurai Pandian
+ * @author Urvil Joshi
+ * @author Srinivasan
+ *
+ */
+@Profile("local")
+@Service
+public class ProxyAuthServiceImpl implements AuthService {
+
+	private static final String CLIENTID_AND_TOKEN_COMBINATION_HAD_BEEN_VALIDATED_SUCCESSFULLY = "Clientid and Token combination had been validated successfully";
+
+	private static final String LOG_OUT_FAILED = "log out failed";
+
+	private static final String FAILED = "Failed";
+
+	private static final String SUCCESS = "Success";
+
+	private static final String SUCCESSFULLY_LOGGED_OUT = "successfully loggedout";
+
+	@Autowired
+	private ProxyTokenGenerator proxyTokenGenarator;
+
+	@Value("${mosip.kernel.open-id-url}")
+	private String keycloakOpenIdUrl;
+
+	@Value("${mosip.admin_realm_id}")
+	private String realmId;
+
+	@Autowired
+	UserStoreFactory userStoreFactory;
+
+	@Autowired
+	KeycloakImpl keycloakImpl;
+
+	@Autowired
+	TokenGenerator tokenGenerator;
+
+	@Autowired
+	TokenValidator tokenValidator;
+
+	@Autowired
+	TokenService customTokenServices;
+
+	@Autowired
+	OTPService oTPService;
+
+	@Autowired
+	UinService uinService;
+
+	@Autowired
+	MosipEnvironment mosipEnvironment;
+
+	@Autowired
+	ObjectMapper objectmapper;
+
+	@Autowired
+	private RestTemplate restTemplate;
+
+	@Value("${mosip.kernel.open-id-url}")
+	private String openIdUrl;
+
+	@Value("${mosip.admin.login_flow.name}")
+	private String loginFlowName;
+
+	@Value("${mosip.admin.clientid}")
+	private String clientID;
+
+	@Value("${mosip.admin.clientsecret}")
+	private String clientSecret;
+
+	@Value("${mosip.admin.redirecturi}")
+	private String redirectURI;
+
+	@Value("${mosip.admin.login_flow.scope}")
+	private String scope;
+
+	@Value("${mosip.admin.login_flow.response_type}")
+	private String responseType;
+
+	@Value("${mosip.keycloak.authorization_endpoint}")
+	private String authorizationEndpoint;
+
+	@Value("${mosip.keycloak.token_endpoint}")
+	private String tokenEndpoint;
+
+	@Value("${mosip.admin_realm_id}")
+	private String realmID;
+
+	@Value("${spring.profiles.active}")
+	String activeProfile;
+
+	@Value("${auth.local.exp:1000000}")
+	long localExp;
+
+	@Value("${auth.local.secret:secret}")
+	String localSecret;
+
+	@Qualifier("authRestTemplate")
+	@Autowired
+	private RestTemplate authRestTemplate;
+
+	/**
+	 * Method used for validating Auth token
+	 * 
+	 * @param token token
+	 * 
+	 * @return mosipUserDtoToken is of type {@link MosipUserTokenDto}
+	 * 
+	 * @throws Exception exception
+	 * 
+	 */
+
+	@Override
+	public MosipUserTokenDto validateToken(String token) throws Exception {
+		MosipUserTokenDto mosipUserDtoToken = tokenValidator.validateToken(token);
+		AuthToken authToken = customTokenServices.getTokenDetails(token);
+		if (authToken == null) {
+			throw new AuthManagerException(AuthErrorCode.INVALID_TOKEN.getErrorCode(),
+					AuthErrorCode.INVALID_TOKEN.getErrorMessage());
+		}
+		if (mosipUserDtoToken != null /* && (currentTime < authToken.getExpirationTime()) */) {
```

Comment:
Please uncomment the expiration check. I assume this is a proxy class and used only in the local profile. The expiration scenarios will need to be tested from the local. So please uncomment.

Reply:
This method is not even used by our controller . This is code of old IAM which was there as a part of migration strategy which will be removed after this  version. Our team is working on code cleanup which will be done in next build.